mediagrains/gsf.py

```diff
@@ -339,24 +339,29 @@ def _decode_gbhd(self, b, i):
                 (meta['grain']['cog_frame']['height'], i) = self._read_uint(b, i, 4)
                 (meta['grain']['cog_frame']['extension'], i) = self._read_uint(b, i, 4)
                 (ar, i) = self._read_rational(b, i)
-                meta['grain']['cog_frame']['source_aspect_ratio'] = {'numerator': ar.numerator,
-                                                                     'denominator': ar.denominator}
+                if ar != 0:
+                    meta['grain']['cog_frame']['source_aspect_ratio'] = {'numerator': ar.numerator,
+                                                                         'denominator': ar.denominator}
                 (ar, i) = self._read_rational(b, i)
-                meta['grain']['cog_frame']['pixel_aspect_ratio'] = {'numerator': ar.numerator,
-                                                                    'denominator': ar.denominator}
+                if ar != 0:
+                    meta['grain']['cog_frame']['pixel_aspect_ratio'] = {'numerator': ar.numerator,
+                                                                        'denominator': ar.denominator}
                 meta['grain']['cog_frame']['components'] = []
                 if i < block_end:
                     comp_start = i
                     (tag, size, i) = self._decode_block_header(b, i, ["comp"], optional=True)
                     comp_end = comp_start + size
                     if size != 0:
                         (n_comps, i) = self._read_uint(b, i, 2)
+                        offset = 0
                         for c in range(0, n_comps):
                             comp = {}
                             (comp['width'], i) = self._read_uint(b, i, 4)
                             (comp['height'], i) = self._read_uint(b, i, 4)
                             (comp['stride'], i) = self._read_uint(b, i, 4)
                             (comp['length'], i) = self._read_uint(b, i, 4)
+                            comp['offset'] = offset
```

Comment:
I assume this got through originally because there isn't a test to exercise it - could you add one?